### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by rn**
- Build using `docker`
- Make if work on Windows (use MSYS `bash.exe`)
- `RT_LABELS` is supposed to be a list of the run time labels and is supposed to be colon separated. Otherwise script functions like `rt_label_set()` don't work.
- `RT_ROOT` is supposed to point to the root of the `rtf` source tree to pick up the library and the utilities. It defaults to the go path but we now allow overriding it via a environment variable.
- `rtf` was executing the group `init` and `deinit` functions even if no tests in a group where selected. Fix it.
- Fix version on macOS

### CODE REVIEW

**Comment by rn** [comment]
pushed an update which adds a `build-with-docker` target in addition to the existing ones

File: Makefile [hunk 1]
```diff
@@ -16,6 +25,17 @@ INEFFASSIGN:=$(shell command -v ineffassign 2> /dev/null)
 
 default: rtf
 
+# Build with docker
+GO_COMPILE=linuxkit/go-compile:7cac05c5588b3dd6a7f7bdb34fc1da90257394c7
+build-with-docker: tmp_rtf_bin.tar
+	tar xf $<
+	rm $<
+
+tmp_rtf_bin.tar: $(_DEPS)
```
Comment (dave-tucker) [inline Makefile]: Should this be `$(DEPS)`?

File: Makefile [hunk 2]
```diff
@@ -16,6 +25,17 @@ INEFFASSIGN:=$(shell command -v ineffassign 2> /dev/null)
 
 default: rtf
 
+# Build with docker
+GO_COMPILE=linuxkit/go-compile:7cac05c5588b3dd6a7f7bdb34fc1da90257394c7
+build-with-docker: tmp_rtf_bin.tar
```
Comment (dave-tucker) [inline Makefile]: This target should be `PHONY`

**Comment by rn** [comment]
hopefully now fixed the makefile

**Pull request merged**